File: missing_integer/missing_integer.js
// https://app.codility.com/demo/results/trainingC5NWNY-Q9W/

function solution(A) {
  let result = 1;
  const sorted = A.sort((a, b) => a - b);
  const setSorted = new Set(sorted);

  for (let i = 1; i <= setSorted.size; i++) {
    if (!setSorted.has(result)) {
      break;
    } else {
      result++;
    }
  }

  return result;
}

File: missing_integer/missing_integer.py
# https://app.codility.com/demo/results/trainingMKFNH5-M5Z/


def solution(A):
    occurrences_map = [False] * (len(A) + 1)

    for number in A:
        if (1 <= number <= len(A) + 1):
            occurrences_map[number - 1] = True

    result = 1

    for number in range(len(A) + 1):
        if (not occurrences_map[number]):
            result = number + 1
            break

    return result
